addon/utils/object-transforms.ts

```diff
@@ -0,0 +1,56 @@
+import { isPresent } from '@ember/utils';
+
+type Options = Record<string, string>;
+
+/**
+ * Present config options
+ * @param {Options} opts
+ * @returns {Options}
+ */
+export function compact(opts: Options): Options {
+  return includeKeys(opts, (key) => isPresent(opts[key]));
+}
+
+/**
+ * Config options without `excludedKeys`
+ * @param {Options} opts
+ * @param {string[]} excludedKeys
+ * @returns {Options}
+ */
+export function without(opts: Options, excludedKeys: string[]): Options {
+  return includeKeys(opts, (key) => !excludedKeys.includes(key));
+}
+
+/**
+ * Config options with allowed keys
+ * @param {Options} opts
+ * @param {(key:string) => boolean} include
+ * @returns {Options}
+ */
+function includeKeys(
+  opts: Options,
+  include: (key: string) => boolean
+): Options {
+  const newOpts: Options = {};
+
+  for (const key in opts) {
+    if (include(key)) newOpts[key] = opts[key];
+  }
+
+  return newOpts;
+}
+
+/**
+ * Check if given object has any own properties
+ * @param {Record<string,unknown>} obj
+ * @returns {boolean}
+ */
+export function hasKeys(obj: Record<string, unknown>): boolean {
```

Comment:
I'm struggling a little with this function's name. From its name I expected it to check if a given object has a key, or keys, as one of its properties. Would `hasOwnProperties` be more clear?

Reply:
Yeah I agree that reads better and feels closer to the browser API.